pkg/controller/vm_webhook.go

```diff
@@ -76,7 +79,34 @@ func MutateVM(ctx context.Context, vm *virtv1alpha1.VirtualMachine, oldVM *virtv
 	}
 
 	if vm.Spec.Instance.Memory.Size == nil {
-		vm.Spec.Instance.Memory.Size = vm.Spec.Resources.Requests.Memory()
+		if !vm.Spec.Resources.Requests.Memory().IsZero() {
+			vm.Spec.Instance.Memory.Size = vm.Spec.Resources.Requests.Memory()
+		} else {
+			vm.Spec.Instance.Memory.Size = func() *resource.Quantity { q := resource.MustParse("1Gi"); return &q }()
+		}
+	}
+
+	if vm.Spec.Instance.Memory.Hugepages != nil {
+		hugepagesSize := fmt.Sprintf("hugepages-%s", vm.Spec.Instance.Memory.Hugepages.PageSize)
+
+		if vm.Spec.Resources.Limits == nil {
+			vm.Spec.Resources.Limits = corev1.ResourceList{}
+		}
+		hugepagesLimit, exist := vm.Spec.Resources.Limits[corev1.ResourceName(hugepagesSize)]
+		if !exist {
+			hugepagesLimit = vm.Spec.Instance.Memory.Size.DeepCopy()
+			vm.Spec.Resources.Limits[corev1.ResourceName(hugepagesSize)] = hugepagesLimit
+		}
+		if vm.Spec.Resources.Requests == nil {
+			vm.Spec.Resources.Requests = corev1.ResourceList{}
+		}
+		if _, exist := vm.Spec.Resources.Requests[corev1.ResourceName(hugepagesSize)]; !exist {
+			vm.Spec.Resources.Requests[corev1.ResourceName(hugepagesSize)] = hugepagesLimit.DeepCopy()
+		}
+
+		if vm.Spec.Resources.Limits.Cpu().IsZero() && vm.Spec.Resources.Limits.Memory().IsZero() && vm.Spec.Resources.Requests.Cpu().IsZero() && vm.Spec.Resources.Requests.Memory().IsZero() {
+			vm.Spec.Resources.Requests[corev1.ResourceMemory] = resource.MustParse(memoryOverhead)
```

Comment:
memory request = memory overhead?

Reply:
Yes, the memory request here means memory overhead except hugepages, and hugepages is reuqested by `hugepages-xxx: n`.